File: pe-gui/src/main/java/org/saar/gui/style/property/Position.java
package org.saar.gui.style.property;

import org.joml.Vector2i;
import org.joml.Vector2ic;

public class Position {

    public final Coordinate x;
    public final Coordinate y;

    private final Vector2i vector = new Vector2i();

    public Position(Coordinate x, Coordinate y) {
        this.x = x;
        this.y = y;
    }

    public void set(Position position) {
        this.x.set(position.x);
        this.y.set(position.y);
    }

    public Vector2ic asVector2i() {
        final int x = this.x.get();
        final int y = this.y.get();
        return this.vector.set(x, y);
    }

}

File: pe-gui/src/main/kotlin/org/saar/gui/style/position/Position.kt
package org.saar.gui.style.position

import org.saar.gui.UIChildNode

class Position(private val container: UIChildNode, default: PositionValue = PositionValues.relative) : ReadonlyPosition {

    override var value: PositionValue = default

    override fun getX() = this.value.computeAxisX(this.container)

    override fun getY() = this.value.computeAxisY(this.container)

}